btf/ext_info.go

```diff
@@ -111,6 +111,16 @@ func (ei *ExtInfos) Assign(insns asm.Instructions, section string) {
 	lineInfos := ei.lineInfos[section]
 	reloInfos := ei.relocationInfos[section]
 
+	AssignMetadataToInstructions(insns, funcInfos, lineInfos, reloInfos)
+}
+
+// Assign per-instruction metadata to the instructions in insns.
```

Comment:
Did this. But I also realized that since offset is a non-exported value at the moment, any users outside of the btf package will have no way to actually sort. Would it make sense to define a type alias for `[]LineInfo` and `[]FuncInfo` which implement [`sort.Interface`](https://pkg.go.dev/sort#Interface)?

Reply:
Hmm good point. New types with `sort.Interface` kind of make sense, except that I don't expect users to ever need to sort the slices, since they're sorted by the constructor. Doing it this way might lead to over sorting.

Thinking about a future API extensions: appending two `[]FuncInfo` to each other doesn't produce a valid result, since the instruction offsets are all wrong. So if we ever want to support that we'd need a wrapper type. So for now, I'd drop the documentation here. Export `type FuncInfos struct { infos []funcInfo }`, etc. instead of `FuncInfo`. That prevents users from messing with the order. Somewhere down the line we can add `FuncInfos.Append(FuncInfos)` if necessary.

What do you think?

Reply:
Sounds like a good idea :+1: